src/pybind/mgr/cephadm/services/nfs.py

```
@@ -29,12 +29,21 @@ def config(self, spec: NFSServiceSpec) -> None:
 
     def prepare_create(self, daemon_spec: CephadmDaemonSpec[NFSServiceSpec]) -> CephadmDaemonSpec:
         assert self.TYPE == daemon_spec.daemon_type
+        # if spec is not attached to daemon_spec it is likely a redeploy or reconfig and
+        # spec should be in spec store
```

can we please fail hard, if we don't have a spec here?

---

I mean, we can't properly create the spec out of thin air.